### ISSUE TITLE
Container fails initial restart, succeeds on second

### ISSUE THREAD

**Issue by michaeldavie** [pi-hole/docker-pi-hole#279]
This is a... 

- [ ] Request for a new or modified feature
- [x] Issue trying to run the docker image
- [ ] Issue trying to build / test / develop the docker image

## Description

When Container Linux restarts, usually due to an update of the host OS, it attempts to restart the Pi-Hole container. However, the container does not load properly until it is restarted manually.

## Expected Behavior

Pi-Hole container restarts automatically.

## Actual Behavior

Pi-Hole container must be restarted manually.

## Steps to Reproduce and debugging done

```
docker run -d \
    --name="home-assistant" \
    --restart=always \
    -v /home/core/home-assistant:/config \
    -e TZ=America/Toronto \
    --net=host \
    homeassistant/home-assistant:latest
```
I've confirmed that this issue occurs when the host OS is restarted, and that it is resolved if I manually restart the container with `docker restart`.

## Debug steps I have tried

- [ ] I have tried destroying my container instance, pulling the newest image version, and re-creating a new container
- [x] I have tried running the nearly stock `docker run` example in the readme (removing any customizations I added)
- [ ] I have tried running without my volume data mounts to eliminate volumes as the cause
- [x] I have searched this repository for existing issues and pull requests that look similar 

## Context and extra information

I'm not sure where to look for additional logs that could help to troubleshoot this issue.

## Your Environment 

* Docker Host Operating System and OS Version: Container Linux 1745.5.0
* Docker Version: 18.03.1-ce
* Hypervisor: VMWare Fusion 8.5.10
* Hardware architecture: Mac mini, Intel Core i5

**Comment by michaeldavie** [comment]
This issue recurred with today's upgrade of Container Linux. @diginc, any guidance on how I could help to troubleshoot would be appreciated. Thanks.

**Comment by diginc** [comment]
Did you mean to post the run command of `homeassistant/home-assistant:latest` instead of a pihole container?

There's a new image available now with Pi-hole v4.0 : `pihole/pihole:latest` which I'd encourage you to try - be sure to review the updated [README](https://github.com/pi-hole/docker-pi-hole/blob/master/README.md)

Sorry for the late response.

**Comment by michaeldavie** [comment]
No worries, thanks for putting this together. Yeah, that's clearly the wrong config...my bad. It seems to have been okay lately with v3.3.1, but I'll switch to v4 and let you know if there are any issues.
